### PR TITLE
[buck-yeh-bux-sqlite] New port

### PR DESCRIPTION
**Pull request by buck-yeh**
The idea is to minimize boilerplate code when doing the same things using native [SQLite API](https://www.sqlite.org/capi3ref.html). Only recurring usages are wrapped into helper classes or plain functions. Mixed uses of this module and the native API are expected. Always prefer `bux::C_SQLite`, `bux::C_SQLiteStmt` over `sqlite3*`, `sqlite3_stmt*` and you will be fine.

- #### What does your PR fix?
  Nothing

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Supported
  - x64-windows
  - x64-windows-static
  - x64-windows-static-md
  - x86-windows
  - x86-windows-static
  - x86-windows-static-md
 
  Yes, `buck-yeh-bux-sqlite:x64-linux=fail` is inserted.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -61,6 +61,7 @@ botan:x64-uwp=fail
 breakpad:arm64-windows=fail
 buck-yeh-bux:x64-linux=fail
 buck-yeh-bux-mariadb-client:x64-linux=fail
+buck-yeh-bux-sqlite:x64-linux=fail
```
Comment (autoantwort) [inline scripts/ci.baseline.txt:64]: If it is not supported you should use `"supports": "!(arm | uwp | osx | linux)"`

File: ports/buck-yeh-bux-mariadb-client/vcpkg.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "Loose-coupled throw-on-error C++20 wrapper classes and utilities over mysql/mariadb Connector/C API",
```
Comment (Cheney-W) [inline ports/buck-yeh-bux-mariadb-client/vcpkg.json:4]:
```suggestion
  "port-version": 1,
  "description": "Loose-coupled throw-on-error C++20 wrapper classes and utilities over mysql/mariadb Connector/C API",
```

File: ports/buck-yeh-bux/vcpkg.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "description": "A supplemental C++ library with functionalities not directly supported from Modern C++ standard.",
```
Comment (Cheney-W) [inline ports/buck-yeh-bux/vcpkg.json:4]:
```suggestion
  "port-version": 1,
  "description": "A supplemental C++ library with functionalities not directly supported from Modern C++ standard.",
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]
`port-version` is different with the version of the underlying library, due to you made a change to this port, so this shouls add `port-version`. You can see doc about `port-version`: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#port-version

**Comment by buck-yeh** [comment]
**arm64_windows failed** ? 
I don't understand. The ports have all been explicitly excluded arm support from.

**Review by JackBoosY — APPROVED** [review]

**Review by Cheney-W — APPROVED** [review]

**Comment by dg0yt** [comment]
This is another package with no indication of usage outside upstream.

I also find it strange that upstream supports linux, but the port doesn't.

**Comment by JackBoosY** [comment]
`buck-yeh-bux-mariadb-client:x64-linux:  cascade: 481f924de6029a657d8befff5c61d01233c22bccbf6aa6cad9683d88549f6d64` is `cascade` since we skipped `libmariadb:x64-linux`.

**Comment by Cheney-W** [comment]
Verified in my local VM, following ports installed successfully:
```
buck-yeh-bux:x64-windows
buck-yeh-bux-mariadb-client:x64-windows
buck-yeh-bux-sqlite:x64-windows
```

**Comment by ras0219-msft** [comment]
Thanks for the new library!

**Pull request merged**